src/module.rs

```diff
@@ -85,6 +86,58 @@ where
   }
 }
 
+// System V AMD64 ABI: Local<Module> returned in a register.
+#[cfg(not(target_os = "windows"))]
+pub type ResolveModuleCallback<'a> = extern "C" fn(
+  Local<'a, Context>,
+  Local<'a, String>,
+  Local<'a, FixedArray>,
+  Local<'a, Module>,
+) -> *const Module;
+
+// Windows x64 ABI: Local<Module> returned on the stack.
+#[cfg(target_os = "windows")]
+pub type ResolveModuleCallback<'a> = extern "C" fn(
+  *mut *const Module,
+  Local<'a, Context>,
+  Local<'a, String>,
+  Local<'a, FixedArray>,
+  Local<'a, Module>,
+) -> *mut *const Module;
+
+impl<'a, F> MapFnFrom<F> for ResolveModuleCallback<'a>
+where
+  F: UnitType
+    + Fn(
+      Local<'a, Context>,
+      Local<'a, String>,
+      Local<'a, FixedArray>,
+      Local<'a, Module>,
+    ) -> Option<Local<'a, Module>>,
+{
+  #[cfg(not(target_os = "windows"))]
+  fn mapping() -> Self {
+    let f = |context, specifier, import_assertions, referrer| {
+      (F::get())(context, specifier, import_assertions, referrer)
+        .map(|r| -> *const Module { &*r })
+        .unwrap_or(null())
+    };
+    f.to_c_fn()
+  }
+
+  #[cfg(target_os = "windows")]
+  fn mapping() -> Self {
+    let f = |ret_ptr, context, specifier, import_assertions, referrer| {
+      let r = (F::get())(context, specifier, import_assertions, referrer)
+        .map(|r| -> *const Module { &*r })
+        .unwrap_or(null());
+      unsafe { std::ptr::write(ret_ptr, r) }; // Write result to stack.
+      ret_ptr // Return stack pointer to the return value.
```

Comment:
IMO, these comments are kind of confusing. They suggest we're directly manipulating the stack but in fact we're writing to the storage allocated by our caller, and that _may_ be on the stack.

`s/stack/caller-allocated memory/` and all should be good. :-)

Reply:
That's a copy-paste from `ResolveCallback` written by Bert, I can update them tho.